src/components/hooks/useForm/useForm.test.ts

```diff
@@ -0,0 +1,211 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import useForm from './useForm';
+
+interface TTestForm {
+  name: string;
+  lastName: string;
+}
+
+const initialValues = {
+  name: 'name',
+  lastName: 'lastName',
+};
+const errorMessage = 'required';
+const onSubmit = jest.fn();
+const validate = (values: TTestForm) => {
+  const errors: Partial<TTestForm> = {};
+  if (values.name === '') {
+    errors.name = errorMessage;
+  }
+  if (values.lastName === '') {
+    errors.lastName = errorMessage;
+  }
+  return errors;
+};
+
+describe('useForm', () => {
+  afterEach(() => {
+    onSubmit.mockClear();
+  });
+
+  it('should return the correct object', () => {
+    const { result } = renderHook(() =>
+      useForm({
+        initialValues,
+        onSubmit,
+        validate,
```

Comment:
there's a type error here:
```
Error:(36, 9) TS2322: Type '(values: TTestForm) => Partial<TTestForm>' is not assignable to type 'TValidate'.
  Types of parameters 'values' and 'values' are incompatible.
    Type 'TDictionary<any>' is missing the following properties from type 'TTestForm': name, lastName
```

Reply:
modified types and that shouldn't happen anymore. Also, I'm gonna be looking at improving them